Fix failing publish errors and publish

v0.7.0 [publish](https://github.com/novifinancial/akd/runs/7457531480?check_suite_focus=true#step:5:48) failed at dry run with errors:

```
error[E0437]: type `StorageKey` is not a member of trait `Storable`
  --> src/append_only_zks.rs:48:5
   |
48 |     type StorageKey = u8;
   |     ^^^^^^^^^^^^^^^^^^^^^ not a member of trait `Storable`

error[E0437]: type `StorageKey` is not a member of trait `crate::storage::Storable`
   --> src/storage/types.rs:122:5
    |
122 |     type StorageKey = ValueStateKey;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `crate::storage::Storable`

error[E0437]: type `StorageKey` is not a member of trait `Storable`
  --> src/tree_node.rs:87:5
   |
87 |     type StorageKey = NodeKey;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `Storable`

error[E0220]: associated type `StorageKey` not found for `St`
   --> src/storage/mod.rs:107:48
    |
107 |     async fn get<St: Storable>(&self, id: &St::StorageKey) -> Result<DbRecord, StorageError>;
    |                                                ^^^^^^^^^^ associated type `StorageKey` not found

error[E0220]: associated type `StorageKey` not found for `St`
   --> src/storage/mod.rs:110:55
    |
110 |     async fn get_direct<St: Storable>(&self, id: &St::StorageKey)
    |                                                       ^^^^^^^^^^ associated type `StorageKey` not found

error[E0220]: associated type `StorageKey` not found for `St`
   --> src/storage/mod.rs:126:20
    |
126 |         ids: &[St::StorageKey],
    |                    ^^^^^^^^^^ associated type `StorageKey` not found

error[E0220]: associated type `StorageKey` not found for `St`
   --> src/storage/timed_cache.rs:127:58
    |
127 |     pub async fn hit_test<St: Storable>(&self, key: &St::StorageKey) -> Option<DbRecord> {
    |                                                          ^^^^^^^^^^ associated type `StorageKey` not found

error[E0220]: associated type `StorageKey` not found for `St`
   --> src/storage/transaction.rs:151:53
    |
151 |     pub async fn get<St: Storable>(&self, key: &St::StorageKey) -> Option<DbRecord> {
    |                                                     ^^^^^^^^^^ associated type `StorageKey` not found

error[E0220]: associated type `StorageKey` not found for `St`
   --> src/storage/memory.rs:157:48
    |
157 |     async fn get<St: Storable>(&self, id: &St::StorageKey) -> Result<DbRecord, StorageError> {
    |                                                ^^^^^^^^^^ associated type `StorageKey` not found

error[E0220]: associated type `StorageKey` not found for `St`
   --> src/storage/memory.rs:170:18
    |
170 |         id: &St::StorageKey,
    |                  ^^^^^^^^^^ associated type `StorageKey` not found

error[E0220]: associated type `StorageKey` not found for `St`
   --> src/storage/memory.rs:206:20
    |
206 |         ids: &[St::StorageKey],
    |                    ^^^^^^^^^^ associated type `StorageKey` not found
```

We should fix these and publish the latest version (will need a bump).

--- a/akd/src/storage/mod.rs
+++ b/akd/src/storage/mod.rs
@@ -58,14 +58,14 @@ pub trait Storable: Clone + Serialize + DeserializeOwned + Sync {
 #[cfg(not(feature = "serde_serialization"))]
 pub trait Storable: Clone + Sync {
     /// This particular storage will have a key type
-    type Key: Clone + Eq + Hash + Send + Sync + std::fmt::Debug;
+    type StorageKey: Clone + Eq + Hash + Send + Sync + std::fmt::Debug;
 
     /// Must return a valid storage type
     fn data_type() -> StorageType;
 
     /// Retrieve an instance of the id of this storable. The combination of the
     /// storable's StorageType and this id are _globally_ unique
-    fn get_id(&self) -> Self::Key;
+    fn get_id(&self) -> Self::StorageKey;
 
     /// Retrieve the full binary version of a key (for comparisons)
     fn get_full_binary_id(&self) -> Vec<u8> {
@@ -73,10 +73,10 @@ pub trait Storable: Clone + Sync {
     }
 
     /// Retrieve the full binary version of a key (for comparisons)
-    fn get_full_binary_key_id(key: &Self::Key) -> Vec<u8>;
+    fn get_full_binary_key_id(key: &Self::StorageKey) -> Vec<u8>;
 
     /// Reformat a key from the full-binary specification
-    fn key_from_full_binary(bin: &[u8]) -> Result<Self::Key, String>;
+    fn key_from_full_binary(bin: &[u8]) -> Result<Self::StorageKey, String>;
 }
 
 /// Storage layer with support for asynchronous work and batched operations

--- a/akd/src/tree_node.rs
+++ b/akd/src/tree_node.rs
@@ -146,6 +146,7 @@ impl TreeNodeWithPreviousValue {
     /// Construct a TreeNode with "previous" value where the
     /// previous value is None. This is useful for the first
     /// time a node appears in the directory data layer.
+    #[cfg(feature = "public-tests")]
     pub(crate) fn from_tree_node(node: TreeNode) -> Self {
         Self {
             label: node.label,

